tests/Unit/CoordinateTest.php

```diff
@@ -556,104 +555,14 @@ public function dataProvider96Well(): array
-        self::assertSame($columnFlowPosition, $coordinate->position(FlowDirection::COLUMN()));
-        self::assertSame($rowFlowPosition, $coordinate->position(FlowDirection::ROW()));
-    }
-
-    /**
-     * @return list<array{row: string, column: int, rowFlowPosition: int, columnFlowPosition: int}>
-     */
-    public function dataProvider12Well(): array
-    {
-        return [
-            [
-                'row' => 'A',
-                'column' => 1,
-                'rowFlowPosition' => 1,
-                'columnFlowPosition' => 1,
-            ],
-            [
-                'row' => 'A',
-                'column' => 2,
-                'rowFlowPosition' => 2,
-                'columnFlowPosition' => 4,
-            ],
-            [
-                'row' => 'A',
-                'column' => 3,
-                'rowFlowPosition' => 3,
-                'columnFlowPosition' => 7,
-            ],
-            [
-                'row' => 'A',
-                'column' => 4,
-                'rowFlowPosition' => 4,
-                'columnFlowPosition' => 10,
-            ],
-            [
-                'row' => 'B',
-                'column' => 1,
-                'rowFlowPosition' => 5,
-                'columnFlowPosition' => 2,
-            ],
-            [
-                'row' => 'B',
-                'column' => 2,
-                'rowFlowPosition' => 6,
-                'columnFlowPosition' => 5,
-            ],
-            [
-                'row' => 'B',
-                'column' => 3,
-                'rowFlowPosition' => 7,
-                'columnFlowPosition' => 8,
-            ],
-            [
-                'row' => 'B',
-                'column' => 4,
-                'rowFlowPosition' => 8,
-                'columnFlowPosition' => 11,
-            ],
-            [
-                'row' => 'C',
-                'column' => 1,
-                'rowFlowPosition' => 9,
-                'columnFlowPosition' => 3,
-            ],
-            [
-                'row' => 'C',
-                'column' => 2,
-                'rowFlowPosition' => 10,
-                'columnFlowPosition' => 6,
-            ],
-            [
-                'row' => 'C',
-                'column' => 3,
-                'rowFlowPosition' => 11,
-                'columnFlowPosition' => 9,
-            ],
-            [
-                'row' => 'C',
-                'column' => 4,
-                'rowFlowPosition' => 12,
-                'columnFlowPosition' => 12,
-            ],
-        ];
-    }
-
     /**
      * @dataProvider invalidRowsOrColumns
      */
     public function testThrowsOnInvalidRowsOrColumns(string $row, int $column): void
     {
         $this->expectException(InvalidArgumentException::class);
-        new Coordinate($row, $column, new CoordinateSystem96Well());
+        $coordinate = new Coordinate($row, $column);
+        $coordinate->validateAgainst(new CoordinateSystem96Well());
```

Comment:
This should be enforced during construction by parameterizing the Coordinate with the type of Coordinate system.